metrics-operator/controllers/common/providers/provider.go

```diff
@@ -23,7 +23,7 @@ type KeptnSLIProvider interface {
 	FetchAnalysisValue(ctx context.Context, query string, spec metricsapi.AnalysisSpec, provider *metricsapi.KeptnMetricsProvider) (string, error)
 }
 
-type ProviderFactory func(providerType string, log logr.Logger, k8sClient client.Client) (KeptnSLIProvider, error)
+type NewFactory func(providerType string, log logr.Logger, k8sClient client.Client) (KeptnSLIProvider, error)
```

Comment:
I think here we should keep the name `ProviderFactory` - The name `NewFactory` is confusing to use for a type, as the `New` prefix usually is used for functions that return an instance of a certain type, but not for a type

Reply:
The naming is the biggest problem of software engineering :-) I am ok with the `ProviderFactory`, is @thisthat against it? Let's make the naming clear first before I adapt the code